lib/services/amazon_sns.rb

```diff
@@ -31,17 +30,22 @@ def publish_to_sns(cfg, json)
     begin
       # older version of AWS SDK does not support region configuration
       # http://ruby.awsblog.com/post/TxVOTODBPHAEP9/Working-with-Regions
-      AWS.config(sns_endpoint: "sns.#{cfg['sns_region']}.amazonaws.com")
-      sns = AWS::SNS.new(config(cfg['aws_key'], cfg['aws_secret']))
-      topic = sns.topics[cfg['sns_topic']]
-      topic.publish(json)
-      return sns
-
-    rescue AWS::SNS::Errors::AuthorizationError,
-           AWS::SNS::Errors::InvalidClientTokenId,
-           AWS::SNS::Errors::SignatureDoesNotMatch => e
+      sns = Aws::SNS::Client.new(:region => cfg['sns_region'],
+                                :access_key_id => cfg['aws_key'],
+                                :secret_access_key => cfg['aws_secret'])
+      attrs = {
+        "X-Github-Event" => {data_type: "String",
+                             string_value: event.to_s
+                            }
+      }
+      sns.publish({
+                    message: json,
+                    topic_arn: cfg['sns_topic'],
```

Comment:
The original code was also using topic arn (the config just isn't called topic arn) so there is no change.  It was using:
```ruby
ns = AWS::SNS.new(config(cfg['aws_key'], cfg['aws_secret']))
topic = sns.topics[cfg['sns_topic']]
```
sns.topics is a topic collection indexed by the topic arn - https://docs.aws.amazon.com/AWSRubySDK/latest/AWS/SNS/TopicCollection.html

The config could be called cfg['sns_topic_arn'] to better reflect what it is.